src/components/ResourceCard.jsx

```diff
@@ -0,0 +1,310 @@
+
+      if (newFileName !== fileName) {
+        // We'll need to create a new .md file with a new filename
+        params = {
+          content: base64EncodedContent,
+          pageName: newFileName,
+        };
+
+        // If it is an existing post, delete the existing page
+        if (!isNewPost) {
+          await axios.delete(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/resources/${category}/pages/${fileName}`, {
+            data: {
+              sha,
+            },
+            withCredentials: true,
+          });
+        }
+
+        await axios.post(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/resources/${category}/pages`, params, {
+          withCredentials: true,
+        });
+      } else {
+        // Save to existing .md file
+        params = {
+          content: base64EncodedContent,
+          sha,
+        };
+
+        await axios.post(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/resources/${category}/pages/${fileName}`, params, {
+          withCredentials: true,
+        });
+      }
+
+      // Refresh page
+      window.location.reload();
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  changeHandler = (event) => {
+    const { id, value } = event.target;
+    this.setState({ [id]: value });
+  }
+
+  render() {
+    const {
+      type,
+      category,
+      title,
+      date,
+      siteName,
+      fileName,
+      resourceCategories,
+      isNewPost,
+    } = this.props;
+    const {
+      settingsIsActive,
+      title: stateTitle,
+      category: stateCategory,
+      date: stateDate,
+      permalink: statePermalink,
+      fileUrl: stateFileUrl,
+    } = this.state;
+    return (
+      <div>
+        { isNewPost
+          ? <button type="button" onClick={this.settingsToggle}>Create New Post</button>
+          : (
+            <>
+              <p>
+Category:
+                {prettifyResourceCategory(category)}
+              </p>
+              <p>
+Title:
+                {title}
+              </p>
+              <p>
+Date:
+                {date}
+              </p>
+              <p>
+Type:
+                {type}
+              </p>
+              <button type="button" onClick={this.settingsToggle}>Settings</button>
+            </>
+          )}
+        {settingsIsActive
+          ? (
+            <ResourceCardModal
+              title={stateTitle}
+              category={stateCategory}
+              date={stateDate}
+              permalink={statePermalink}
+              fileUrl={stateFileUrl}
+              resourceCategories={resourceCategories}
+              changeHandler={this.changeHandler}
+              saveHandler={this.saveHandler}
+              deleteHandler={this.deleteHandler}
+              permalinkFileUrlToggle={this.permalinkFileUrlToggle}
+              isNewPost={isNewPost}
+            />
+          )
+          : null}
+        <Link to={`/sites/${siteName}/resources/${category}/pages/${fileName}`}>Edit</Link>
```

Comment:
The backend endpoint does not work, so when you click into `Edit` it does not return the content

Reply:
Deleted